app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Union
+
+
+def cache(func: Callable) -> Callable:
+    cached_data = {}
+
+    def wrapper(*args) -> Union[int, list]:
```

Comment:
what if decorated func will return a dict or tuple ? fix ur annoation for return type )